### PR TITLE
Add IO Counters

### PR DESCRIPTION
**Pull request by oerling**
Adds a counter for total bytes of IO avoided by hitting RAM cache. Adds the per thread IO wait counter to runTimeStats. Adds a process-wide counter for tracking table scan IO wait.

### CODE REVIEW

File: velox/exec/TableScan.h [hunk 1]
```diff
@@ -101,5 +108,12 @@ class TableScan : public SourceOperator {
 
   // String shown in ExceptionContext inside DataSource and LazyVector loading.
   std::string debugString_;
+
+  // The last value of the IO wait time of 'this' that has been added to the
+  // global static 'ioWaitNanos_'.
+  uint64_t lastIoWaitNanos_{0};
+
+  // Process-wide IO wait time.
+  static uint64_t ioWaitNanos_;
```
Comment (pedroerp) [inline velox/exec/TableScan.h:117]: doesn't this need to be atomic?

File: velox/dwio/common/IoStatistics.h [hunk 2]
```diff
@@ -48,6 +48,11 @@ class IoCounter {
     return bytes_;
   }
 
+  /// Alias of 'bytes', not some counters count different quantities, e.g. time.
+  uint64_t total() const {
+    return bytes_;
```
Comment (Yuhta) [inline velox/dwio/common/IoStatistics.h:53]: Maybe consider renaming `bytes_` to `amount_`

**Review by Yuhta — APPROVED** [review]
@oerling LGTM except two small improvement:
1. Consider renaming `IoCounter::bytes_` to `amount_`
2. Process-wide `ioWaitNanos_` should be atomic

**Comment by Yohahaha** [comment]
Looks we have two metrics for counting IO wait time, `ioWaitNanos_ ` for `CacheBufferedInput`, `totalScanTime` for `BufferedInput`. Can we use define a actual variable to count all the IO wait time?
cc @Yuhta

File: velox/common/caching/AsyncDataCache.cpp [hunk 3]
```diff
@@ -177,6 +177,7 @@ CachePin CacheShard::findOrCreate(
           found->setPrefetch(false);
         } else {
           ++numHit_;
+          hitBytes_ += found->size();
```
Comment (Yohahaha) [inline velox/common/caching/AsyncDataCache.cpp:180]: Why we not count `hitBytes_` if found is prefetch? Data has already loading in memory right? and we actually return the prefetched entry in this branch end.

**Pull request closed without merging**